source/API_Reference/Event_Webhook/event.md

```diff
@@ -0,0 +1,1043 @@
+- <a name="uniqueargs"></a>`unique_args` or `custom_args` - 
+{% anchor h2 %}
+Unique Arguments and Custom Arguments
+{% endanchor %}
+
+Events generated by SendGrid may also include any [unique arguments]({{root_url}}/API_Reference/SMTP_API/unique_arguments.html) or custom arguments.
+
+{% info %}
+Unique arguments and custom arguments are functionally the same: It's called unique arguments in the SMTP API or v2 Mail Send, and it's called custom arguments in v3 Mail Send.
+{% endinfo %}
+
+{% anchor h3 %}
+Unique Arguments
+{% endanchor %}
+
+To define and receive unique arguments when sending email via the [SMTP API]({{root_url}}/API_Reference/SMTP_API/index.html) or the [v2 Mail Send endpoint]({{root_url}}/API_Reference/Web_API/mail.html) simply use the `unique_args` parameter in the X-SMTPAPI header or parameter. For example, if you have an application and want to receive custom parameters such as the `userid` and the email `template`, you would submit them with the X-SMTPAPI header or parameter, as described [here]({{root_url}}/API_Reference/SMTP_API/unique_arguments.html).
+
+For example, if you include the following unique arguments in your x-smtpapi parameter for an email sent via the v2 Mail Send endpoint:
+
+{% codeblock lang:json %}
+{
+  "unique_args": {
+    "userid": "1123",
+    "template": "welcome"
+  }
+}
+{% endcodeblock %}
+
+You will receive the same unique argument included with the data returned to the Event Webhook:
+
+{% codeblock lang:json %}
+[
+  {
+    "sg_message_id":"sendgrid_internal_message_id",
+    "email": "john.doe@sendgrid.com",
+    "timestamp": 1337966815,
+    "event": "click",
+    "url": "https://sendgrid.com",
+    "userid": "1123",
+    "template": "welcome"
+  }
+]
+{% endcodeblock %}
+
+{% warning %}
+You can create unique arguments with the same keys as reserved words, such as "event" or "email". In this case, SendGrid will default to the reserved key and NOT your unique argument. An example of this is below.
+{% endwarning %}
+
+{% anchor h4 %}
+Reserved Keys in Unique Arguments
+{% endanchor %}
+
+{% codeblock lang:json %}
+//for this example, assume we're sending to john.doe@sendgrid.com
+{
+  "unique_args": {
+    "customerAccountNumber": "55555",
+    "activationAttempt": "1",
+    "New Argument 1": "New Value 1",
+    "email": "jane.doe@sendgrid.com",
+    "event": "SendEmail"
+  }
+}
+{% endcodeblock %}
+
+{% anchor h4 %}
+The resulting webhook call
+{% endanchor %}
+
+{% codeblock lang:json %}
+[
+  {
+    "event": "Processed",
+    "timestamp":"123456789",
+    "customerAccountNumber": "55555",
+    "activationAttempt": "1",
+    "New Argument 1": "New Value 1",
+    "email": "john.doe@sendgrid.com"
+  }
+]
+{% endcodeblock %}
+
+*Note: The event and the email do not get overwritten because "event" and "email" are reserved, and SendGrid defaults to the reserved key for webhooks.*
```

Comment:
{% info %} 
The event and the email do not get overwritten because "event" and "email" are reserved, and SendGrid defaults to the reserved key for webhooks
{% endinfo %}